src/sentry/api/endpoints/project_events.py

```diff
@@ -63,3 +49,52 @@ def get(self, request, project):
             on_results=lambda x: serialize(x, request.user),
             paginator_cls=DateTimePaginator,
         )
+
+    def __search_events_snuba(self, request, project):
+        from functools32 import partial
+        from sentry.api.paginator import GenericOffsetPaginator
+        from sentry.api.serializers.models.event import SnubaEvent
+        from sentry.utils.snuba import raw_query
+
+        query = request.GET.get('query')
+        conditions = []
+        if query:
+            conditions.append(['message', 'LIKE', '%{}%'.format(query)])
+
+        data_fn = partial(
+            # extract 'data' from raw_query result
+            lambda *args, **kwargs: raw_query(*args, **kwargs)['data'],
+            start=timezone.now() - timedelta(days=90),
+            end=timezone.now(),
+            conditions=conditions,
+            filter_keys={'project_id': [project.id]},
+            selected_columns=SnubaEvent.selected_columns
+        )
+
+        return self.paginate(
+            request=request,
+            on_results=lambda results: serialize(
+                [SnubaEvent(row) for row in results], request.user),
+            paginator=GenericOffsetPaginator(data_fn=data_fn)
+        )
+
+    @attach_scenarios([list_project_available_samples_scenario])
+    def get(self, request, project):
+        """
+        List a Project's Events
+        ```````````````````````
+
+        Return a list of events bound to a project.
+
+        Note: This endpoint is experimental and may be removed without notice.
+
+        :pparam string organization_slug: the slug of the organization the
+                                          groups belong to.
+        :pparam string project_slug: the slug of the project the groups
+                                     belong to.
+        """
+        backend = request.COOKIES.get('eventstream', 'legacy')
+        return {
+            'legacy': self.__search_events_legacy,
+            'snuba': self.__search_events_snuba,
+        }[backend](request, project)
```

Comment:
This intentionally doesn't deal with the `KeyError` since swapping this backend isn't intended to be a public API (and is temporary.)